payments-core/src/main/java/com/stripe/android/paymentsheet/injection/FlowControllerModule.kt

```diff
@@ -22,107 +21,95 @@ import com.stripe.android.paymentsheet.flowcontroller.DefaultFlowControllerIniti
 import com.stripe.android.paymentsheet.flowcontroller.FlowControllerInitializer
 import com.stripe.android.paymentsheet.flowcontroller.FlowControllerViewModel
 import com.stripe.android.paymentsheet.model.ClientSecret
+import dagger.Binds
 import dagger.Lazy
 import dagger.Module
 import dagger.Provides
 import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.flow.first
 import javax.inject.Named
 import javax.inject.Provider
 import javax.inject.Singleton
 import kotlin.coroutines.CoroutineContext
 
 @Module
-internal class FlowControllerModule {
-    @Provides
-    @Named(ENABLE_LOGGING)
-    fun provideEnabledLogging(): Boolean = false
+internal abstract class FlowControllerModule {
+    @Binds
+    abstract fun bindsFlowControllerInitializer(
+        defaultFlowControllerInitializer: DefaultFlowControllerInitializer
+    ): FlowControllerInitializer
 
-    @Provides
-    @Singleton
-    fun provideLogger(@Named(ENABLE_LOGGING) enableLogging: Boolean) =
-        Logger.getInstance(enableLogging)
+    companion object {
+        @Provides
+        @Named(ENABLE_LOGGING)
+        fun provideEnabledLogging(): Boolean = false
```

Comment:
Unrelated, but if it's easy can we default to `true` when it's a debug build, `false` otherwise?

Reply:
I can look into that in a separate pr